web/src/search.js

```diff
@@ -58,25 +67,47 @@ export function initialize({on_narrow_search}) {
 
     $search_query_box.typeahead({
         source(query) {
+            if (query !== "") {
+                search_input_has_changed = true;
+            }
             const suggestions = search_suggestion.get_suggestions(query);
```

Comment:
Hmm. Structurally, won't `get_suggestions` want to have the existing pills passed as a parameter, so it can do things like "Don't offer DM options if you've already got a `stream` pill"? 

Even if we don't consider it a requirement to implement a full set of such logic in a first version, and thus `search_suggestion.js` just ignores the value, I think that's the right interface here, rather than doing after-the-fact duplicate filtering in this function?

Reply:
Yeah, as I just noted in another comment, I've been working on moving to that strategy. Good callout!